core/src/main/java/com/yahoo/ycsb/workloads/CoreWorkload.java

```diff
@@ -763,5 +764,6 @@ public void doTransactionInsert(DB db)
 
 		HashMap<String, ByteIterator> values = buildValues(dbkey);
 		db.insert(table,dbkey,values);
+		transactioninsertkeysequence.acknowledge(keynum);
```

Comment:
I think the acknowledge(...) should be in a finally. Missing a value will get the AcknowledgedCounterGenerator stuck on the gap in values.

Should we acknowledge the id if the insert is not successful (return != 0)?  If we do then an error on the insert will cause errors to be reported on the update.  Again, if we don't then the current logic in the AcknowledgedCounterGenerator will get stuck on the gap.

Maybe instead of keeping a queue in the AcknowledgedCounterGenerator instead there can just be an AtomicInteger that tracks the "highest" acknowledged value.  The lastInt() returns the value of the AtomicInteger and the acknowledge does a (spin over a) compare and swap if the ```value``` is larger than the current AtomicInteger value.